crates/iceberg-catalog/src/service/authz/implementations/openfga/client.rs

```diff
@@ -83,79 +59,37 @@ pub(crate) async fn new_client_from_config(
 /// - Server connection fails
 /// - Store (name) not found (from crate Config)
 /// - Active Authorization model not found
-pub async fn new_authorizer_from_config() -> OpenFGAResult<Authorizers> {
+pub(crate) async fn new_authorizer_from_config() -> OpenFGAResult<Authorizers> {
     let client = new_client_from_config().await?;
     Ok(Authorizers::OpenFGA(new_authorizer(client, None).await?))
 }
 
 /// Create a new `OpenFGA` authorizer with the given client.
 /// This must be run after migration.
+///
+/// # Errors
+/// - Store does not exist
+/// - Active Authorization model not found
+/// - Server connection fails
+///
+/// # Panics
+/// - If the store is deleted during the runtime of this function
 pub(crate) async fn new_authorizer(
-    mut client: OpenFgaServiceClient<ClientConnection>,
+    mut client: BasicOpenFgaServiceClient,
     store_name: Option<String>,
 ) -> OpenFGAResult<OpenFGAAuthorizer> {
-    let active_model_version = ModelVersion::active();
-    let store_name = store_name.unwrap_or_else(|| AUTH_CONFIG.store_name.clone());
-
-    let store_id = client
+    let store_name = store_name.unwrap_or(AUTH_CONFIG.store_name.clone());
+    let auth_model_id = get_active_auth_model_id(&mut client, Some(store_name.clone())).await?;
+    let store = client
         .get_store_by_name(&store_name)
         .await?
-        .ok_or(OpenFGAError::StoreNotFound { store: store_name })?
-        .id;
-    let authorization_model_id =
-        get_auth_model_id(&mut client, store_id.clone(), active_model_version).await?;
+        .unwrap_or_else(|| panic!("OpenFGA store with name {store_name} exists"));
```

Comment:
do we want to panic here?

Reply:
I made an error from it. Probably cleaner.
Eventually I might pipe it though all the way from openfga_client